File: android/libapp/src/main/java/com/topdon/lms/sdk/LMS.java
package com.topdon.lms.sdk;

import android.content.Context;

/**
 * Stub implementation for LMS SDK
 * 
 * @author Copilot
 */
public class LMS {
    public static final int SUCCESS = 0;
    private static LMS instance = new LMS();
    
    // Context stub
    public static Context mContext;
    
    public static LMS getInstance() {
        return instance;
    }
    
    public String getLoginName() {
        return "guest";
    }
    
    public void checkAppUpdate(CheckUpdateCallback callback) {
        // Stub implementation - create a response object
        CallbackData response = new CallbackData();
        response.code = 2000;
        response.data = "{\"version\":\"1.0.0\"}";
        callback.invoke(response);
    }
    
    public void getStatement(String type, com.topdon.lms.sdk.network.IResponseCallback callback) {
        // Stub implementation
        callback.onResponse("{\"code\":\"2000\",\"data\":{\"url\":\"https://example.com\"}}");
    }
    
    // Inner class for callback data
    public static class CallbackData {
        public int code;
        public String data;
    }
    
    // Functional interface for checkAppUpdate callback
    public interface CheckUpdateCallback {
        void invoke(CallbackData data);
    }
}

File: android/libapp/src/main/java/com/topdon/lms/sdk/LMS.kt
package com.topdon.lms.sdk

/**
 * Stub implementation of LMS SDK for compilation
 * This provides minimal implementation to allow compilation when LMS AAR is in app module
 */
class LMS private constructor() {
    companion object {
        @Volatile
        private var instance: LMS? = null
        
        @JvmStatic
        fun getInstance(): LMS {
            return instance ?: synchronized(this) {
                instance ?: LMS().also { instance = it }
            }
        }
    }
    
    var loginName: String = ""
    
    // Minimal stub methods for compilation
    fun init(context: android.content.Context) {}
    
    fun isLogin(): Boolean = false
    
    fun login(username: String, password: String, callback: Any?) {}
    
    fun checkAppUpdate(callback: Any?) {}
    
    fun getStatement(request: Any?, callback: Any?) {}
}